php/WP_CLI/Runner.php

```diff
@@ -331,30 +331,24 @@ private function _run_command() {
 	}
 
 	/**
-	 * Perform a command against a remote server over SSH
+	 * Perform a command against a remote server over SSH (or a container using
+	 * scheme of "docker" or "docker-compose").
+	 *
+	 * @param string $connection_string Passed connection string.
+	 * @return void
 	 */
-	private function run_ssh_command( $ssh ) {
+	private function run_ssh_command( $connection_string ) {
 
 		WP_CLI::do_hook( 'before_ssh' );
 
-		// host OR host/path/to/wordpress OR host:port/path/to/wordpress
-		$bits = Utils\parse_ssh_url( $ssh );
-		$host = isset( $bits['host'] ) ? $bits['host'] : null;
-		$port = isset( $bits['port'] ) ? $bits['port'] : null;
-		$path = isset( $bits['path'] ) ? $bits['path'] : null;
-
-		WP_CLI::debug( 'SSH host: ' . $host, 'bootstrap' );
-		WP_CLI::debug( 'SSH port: ' . $port, 'bootstrap' );
-		WP_CLI::debug( 'SSH path: ' . $path, 'bootstrap' );
-
-		$is_tty = function_exists( 'posix_isatty' ) && posix_isatty( STDOUT );
+		$bits = Utils\parse_ssh_url( $connection_string );
 
 		$pre_cmd = getenv( 'WP_CLI_SSH_PRE_CMD' );
 		if ( $pre_cmd ) {
 			$pre_cmd = rtrim( $pre_cmd, ';' ) . '; ';
 		}
-		if ( $path ) {
-			$pre_cmd .= "cd {$path}; ";
+		if ( ! empty( $bits['path'] ) ) {
+			$pre_cmd .= 'cd ' . escapeshellarg( $bits['path'] ) . '; ';
```

Comment:
I wonder so many things; I should probably try to cut down. General uninformed feeling:

1. Less code, takes advantage of built-in escaping below.
2. Doing it "the WP-CLI way."
3. Currently, if user supplies path in connection string AND `--path` arg, the `--path` arg "wins" even though it's not clear why or how.

After writing out point 3, though, I realize changing it would be rather arbitrary. If we wanted to be explicit, we could just drop support for path in the connection string and direct users to the `--path` arg.